types/block.go

```diff
@@ -1167,6 +1398,26 @@ func (data *EvidenceData) FromProto(eviData *tmproto.EvidenceList) error {
 	return nil
 }
 
+func (data *EvidenceData) SplitIntoShares() NamespacedShares {
+	rawDatas := make([][]byte, 0, len(data.Evidence))
+	for _, ev := range data.Evidence {
+		pev, err := EvidenceToProto(ev)
+		if err != nil {
+			panic("failure to convert evidence to equivalent proto type")
+		}
+		rawData, err := protoio.MarshalDelimited(pev)
+		if err != nil {
+			panic(err)
+		}
+		rawDatas = append(rawDatas, rawData)
+	}
+	w := NewContiguousShareWriter(consts.EvidenceNamespaceID)
+	for _, evd := range rawDatas {
+		w.Write(evd)
+	}
+	return w.Export()
+}
+
```

Comment:
Orthogonal to this PR: we should consider moving the added methods and types (e.g. ` SplitIntoShares`, `Message ` etc into their own files instead. Such that it becomes more obvious which functionality we've added.

Reply:
I completely agree! This was something that I want to do for v0.36, where we should have almost every change be either a deletion from an existing file, or an addition in a new file. Being relatively strict with that rule will be much more doable when we don't have to change large portions of the boiler plate code for ABCI